lib/prism/parse_result.rb

```diff
@@ -16,16 +16,32 @@ def slice(offset, length)
       source.byteslice(offset, length)
     end
 
+    # Binary search through the offsets to find the line number for the given
+    # offset.
     def line(value)
-      offsets.bsearch_index { |offset| offset > value } || offsets.length
+      left = 0
+      right = offsets.length - 1
+
+      while left <= right
```

Comment:
What's the reason to implement our own binary search instead of using `bsearch_index` here?
`bsearch_index` wasn't correct for some edge case maybe?